notifier/notifier.go

```diff
@@ -160,6 +180,13 @@ func (n *Notifier) ApplyConfig(conf *config.Config) error {
 		if err != nil {
 			return err
 		}
+
+		for _, am := range ams.ams {
```

Comment:
Actually, if I understand the code correctly, no discovery has happened on this new AM set up until this point yet, so `ams.ams` will always be empty. It should be sufficient to only do this in `Sync()`, which actually adds the AM targets.